src/openpgp.js

```diff
@@ -61,19 +61,20 @@ export function generateKey({ userIDs = [], passphrase = "", type = "ecc", rsaBi
   }
   const options = { userIDs, passphrase, type, rsaBits, curve, keyExpirationTime, date, subkeys };
 
-  return generate(options, config).then(async key => {
+  try {
+    const key = await generate(options, config);
     const revocationCertificate = await key.getRevocationCertificate(date, config);
     key.revocationSignatures = [];
 
     return {
-
-      key: key,
+      key,
       privateKeyArmored: key.armor(config),
       publicKeyArmored: key.toPublic().armor(config),
       revocationCertificate: revocationCertificate
-
     };
-  }).catch(onError.bind(null, 'Error generating keypair'));
+  } catch (err) {
+    return onError('Error generating keypair', err);
```

Comment:
It feels weird to me to `return` in a `catch`, maybe we could replace this with
```suggestion
    throw util.wrapError('Error generating keypair', err);
```
\- then it's kind of more obvious what's going on, and we could get rid of `onError` entirely - I don't think the part about `util.printDebugError` is really necessary.
(And similarly everywhere else)

Reply:
Yeah I definitely found the `return onError` quite confusing as well 👍